cardano-api/src/Cardano/Api/SerialiseRaw.hs

```diff
@@ -31,9 +32,25 @@ serialiseToRawBytesHex = Base16.encode . serialiseToRawBytes
 serialiseToRawBytesHexText :: SerialiseAsRawBytes a => a -> Text
 serialiseToRawBytesHexText = Text.decodeUtf8 . serialiseToRawBytesHex
 
-deserialiseFromRawBytesHex :: SerialiseAsRawBytes a
-                           => AsType a -> ByteString -> Maybe a
-deserialiseFromRawBytesHex proxy hex =
-    case Base16.decode hex of
-      Right raw -> deserialiseFromRawBytes proxy raw
-      Left _msg -> Nothing
+data RawBytesHexError
+  = RawBytesHexErrorBase16DecodeFail
+      ByteString -- ^ original input
+      String -- ^ error message
+  | RawBytesHexErrorRawBytesDecodeFail
+      ByteString -- ^ original input
+      -- TODO TypeRep -- ^ output type proxy
+  deriving (Show)
+
+instance Error RawBytesHexError where
+  displayError = \case
+    RawBytesHexErrorBase16DecodeFail input message ->
+      "Base16 cannot decode " ++ show input ++ ": " ++ message
+    RawBytesHexErrorRawBytesDecodeFail input ->
```

Comment:
`"Failed to deserialise: " ++ show input"`

In future (not this PR) we will want to also return a typed error for `SerialiseAsRawBytes`